File: app/src/main/java/com/bookyrself/bookyrself/data/Profile/ProfileDataSource.java
package com.bookyrself.bookyrself.data.Profile;

import com.bookyrself.bookyrself.data.ServerModels.User.User;

import io.reactivex.Flowable;

public interface ProfileDataSource {

    Flowable<User> getProfileInfo(String userId);

    Flowable<User> updateProfileInfo(String userId, User user);

}

File: app/src/main/java/com/bookyrself/bookyrself/data/profile/ProfileDataSource.kt
package com.bookyrself.bookyrself.data.profile

import com.bookyrself.bookyrself.data.serverModels.User.User

import io.reactivex.Flowable

interface ProfileDataSource {

    fun getProfileInfo(userId: String): Flowable<User>

    fun updateProfileInfo(userId: String, user: User): Flowable<User>

}
